examples~/quickstart/client/module_bindings/_Globals/SpacetimeDBClient.cs

```diff
@@ -10,59 +10,94 @@
 	{
-		ReducerType ReducerType { get; }
-		bool InvokeHandler(ReducerEvent reducerEvent);
+		public readonly RemoteTableHandle<EventContext, Message> Message = new();
+		public readonly RemoteTableHandle<EventContext, User> User = new();
 	}
 
-	public partial class ReducerEvent : ReducerEventBase
+	public sealed class RemoteReducers : RemoteBase<DbConnection>
 	{
-		public IReducerArgs? Args { get; }
+		internal RemoteReducers(DbConnection conn) : base(conn) {}
 
-		public string ReducerName => Args?.ReducerName ?? "<none>";
+		public delegate void SendMessageHandler(EventContext ctx, string text);
+		public event SendMessageHandler? OnSendMessage;
 
-		[Obsolete("ReducerType is deprecated, please match directly on type of .Args instead.")]
-		public ReducerType Reducer => Args?.ReducerType ?? ReducerType.None;
+		public void SendMessage(string text)
+		{
+			conn.InternalCallReducer(new SendMessageArgsStruct { Text = text });
+		}
 
-		public ReducerEvent(IReducerArgs? args) : base() => Args = args;
-		public ReducerEvent(TransactionUpdate dbEvent, IReducerArgs? args) : base(dbEvent) => Args = args;
+		public bool InvokeSendMessage(EventContext ctx, SendMessageArgsStruct args)
+		{
+			if (OnSendMessage == null) return false;
+			OnSendMessage(
+				ctx,
+				args.Text
+			);
+			return true;
+		}
+		public delegate void SetNameHandler(EventContext ctx, string name);
+		public event SetNameHandler? OnSetName;
 
-		[Obsolete("Accessors that implicitly cast `Args` are deprecated, please match `Args` against the desired type explicitly instead.")]
-		public SendMessageArgsStruct SendMessageArgs => (SendMessageArgsStruct)Args!;
-		[Obsolete("Accessors that implicitly cast `Args` are deprecated, please match `Args` against the desired type explicitly instead.")]
-		public SetNameArgsStruct SetNameArgs => (SetNameArgsStruct)Args!;
+		public void SetName(string name)
+		{
+			conn.InternalCallReducer(new SetNameArgsStruct { Name = name });
+		}
+
+		public bool InvokeSetName(EventContext ctx, SetNameArgsStruct args)
+		{
+			if (OnSetName == null) return false;
+			OnSetName(
+				ctx,
+				args.Name
+			);
+			return true;
+		}
+	}
+
+	public partial class EventContext : EventContextBase<RemoteTables, RemoteReducers>
+	{
+		public IReducerArgs? Args { get; }
+
+		public string ReducerName => Args?.ReducerName ?? "<none>";
+
+		public EventContext(DbConnection conn, TransactionUpdate update, IReducerArgs? args) : base(conn.RemoteTables, conn.RemoteReducers, update) => Args = args;
 
 		public override bool InvokeHandler() => Args?.InvokeHandler(this) ?? false;
 	}
 
-	public class SpacetimeDBClient : SpacetimeDBClientBase<ReducerEvent>
+	public class DbConnection : DbConnectionBase<DbConnection, EventContext>
 	{
-		protected SpacetimeDBClient()
+		public readonly RemoteTables RemoteTables = new();
+		public readonly RemoteReducers RemoteReducers;
+
+		public DbConnection()
 		{
+			RemoteReducers = new(this);
+
 			clientDB.AddTable<Message>();
```

Comment:
Should the client cache be migrated over to use RemoteTables now?

Reply:
Ah that's a good point.